File: backend/app/schemas/monitoreo.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel


class SesionAccesoOut(BaseModel):
    id: int
    exito: bool
    motivo: Optional[str] = None
    tipo_usuario: Optional[str] = None
    usuario_id: Optional[int] = None
    identificador: Optional[str] = None
    nombres: Optional[str] = None
    apellidos: Optional[str] = None
    rol_codigo: Optional[str] = None
    institucion_educativa_id: Optional[int] = None
    institucion_nombre: Optional[str] = None
    ugel_id: Optional[int] = None
    ugel_nombre: Optional[str] = None
    ip: Optional[str] = None
    dispositivo: Optional[str] = None
    sistema_operativo: Optional[str] = None
    navegador: Optional[str] = None
    es_movil: bool = False
    login_at: Optional[datetime] = None
    last_activity: Optional[datetime] = None
    logout_at: Optional[datetime] = None
    is_active: bool = True
    en_linea: bool = False
    duracion_minutos: Optional[int] = None


class ConteoItem(BaseModel):
    clave: str
    total: int


class PuntoSerie(BaseModel):
    periodo: str
    total: int
    exitos: int = 0
    fallidos: int = 0


class ResumenMonitoreo(BaseModel):
    sesiones_activas: int
    estudiantes_activos: int
    staff_activos: int
    logins_hoy: int
    fallidos_hoy: int
    sesiones_hoy: int
    activos_por_rol: List[ConteoItem]
    activos_por_ugel: List[ConteoItem]
    top_instituciones: List[ConteoItem]
    ultima_actualizacion: datetime


class EstadisticasMonitoreo(BaseModel):
    accesos_por_hora: List[PuntoSerie]
    accesos_por_dia: List[PuntoSerie]
    dispositivos: List[ConteoItem]
    sistemas_operativos: List[ConteoItem]
    navegadores: List[ConteoItem]
    top_instituciones: List[ConteoItem]
    ultima_actualizacion: datetime

File: frontend/src/shared/types/monitoreo.ts
export interface SesionAcceso {
  id: number
  exito: boolean
  motivo: string | null
  tipo_usuario: string | null
  usuario_id: number | null
  identificador: string | null
  nombres: string | null
  apellidos: string | null
  rol_codigo: string | null
  institucion_educativa_id: number | null
  institucion_nombre: string | null
  ugel_id: number | null
  ugel_nombre: string | null
  ip: string | null
  dispositivo: string | null
  sistema_operativo: string | null
  navegador: string | null
  es_movil: boolean
  login_at: string | null
  last_activity: string | null
  logout_at: string | null
  is_active: boolean
  en_linea: boolean
  duracion_minutos: number | null
}

export interface ConteoItem {
  clave: string
  total: number
}

export interface PuntoSerie {
  periodo: string
  total: number
  exitos: number
  fallidos: number
}

export interface ResumenMonitoreo {
  sesiones_activas: number
  estudiantes_activos: number
  staff_activos: number
  logins_hoy: number
  fallidos_hoy: number
  sesiones_hoy: number
  activos_por_rol: ConteoItem[]
  activos_por_ugel: ConteoItem[]
  top_instituciones: ConteoItem[]
  ultima_actualizacion: string
}

export interface EstadisticasMonitoreo {
  accesos_por_hora: PuntoSerie[]
  accesos_por_dia: PuntoSerie[]
  dispositivos: ConteoItem[]
  sistemas_operativos: ConteoItem[]
  navegadores: ConteoItem[]
  top_instituciones: ConteoItem[]
  ultima_actualizacion: string
}

export interface SesionesFiltros {
  page?: number
  size?: number
  q?: string
  rol?: string
  ie_id?: number
  ugel_id?: number
  fecha_desde?: string
  fecha_hasta?: string
  solo_fallidos?: boolean
}
